benchmarks/high_volume_reads_test.go

```diff
@@ -17,126 +18,100 @@ import (
+	Describe("Querying logs from Loki service", func() {
+		BeforeEach(func() {
+			querierDpls = querier.CreateQueriers(highVolumeReadsTest.Readers, benchCfg.Querier)
+			generatorDpl = loadclient.CreateGenerator(highVolumeReadsTest.LogGenerator(), benchCfg.Generator)
 
-		err := k8sClient.Create(context.TODO(), generatorDpl, &client.CreateOptions{})
-		Expect(err).Should(Succeed(), "Failed to deploy logger")
+			err := k8sClient.Create(context.TODO(), generatorDpl, &client.CreateOptions{})
+			Expect(err).Should(Succeed(), "Failed to deploy logger")
 
-		err = utils.WaitForReadyDeployment(k8sClient, generatorDpl, defaultRetry, defaultTimeout)
-		Expect(err).Should(Succeed(), "Failed to wait for ready logger deployment")
+			err = utils.WaitForReadyDeployment(k8sClient, generatorDpl, defaultRetry, defaultTimeout)
+			Expect(err).Should(Succeed(), "Failed to wait for ready logger deployment")
 
-		err = utils.WaitUntilReceivedBytes(metricsClient, scenarioCfgs.StartThreshold, defaultRange, defaultRetry, defaultTimeout)
-		Expect(err).Should(Succeed(), "Failed to wait until latch activated")
+			// Begin loading data into the Loki service so there is something to query for.
+			time.Sleep(time.Minute * 5)
 
-		err = k8sClient.Delete(context.TODO(), generatorDpl, &client.DeleteOptions{})
-		Expect(err).Should(Succeed(), "Failed to delete logger deployment")
-	})
+			for _, dpl := range querierDpls {
+				err := k8sClient.Create(context.TODO(), dpl, &client.CreateOptions{})
+				Expect(err).Should(Succeed(), "Failed to deploy querier")
 
-	for _, scenarioCfg := range scenarioCfgs.Configurations {
-		scenarioCfg := scenarioCfg
-		querierDpls := querier.CreateQueriers(scenarioCfg.Readers, benchCfg.Querier)
+				err = utils.WaitForReadyDeployment(k8sClient, dpl, defaultRetry, defaultTimeout)
+				Expect(err).Should(Succeed(), fmt.Sprintf("Failed to wait for ready querier deployment: %s", dpl.GetName()))
+			}
 
-		Describe(fmt.Sprintf("Configuration: %s", scenarioCfg.Description), func() {
-			BeforeEach(func() {
-				for _, dpl := range querierDpls {
-					err := k8sClient.Create(context.TODO(), dpl, &client.CreateOptions{})
-					Expect(err).Should(Succeed(), "Failed to deploy querier")
+			DeferCleanup(func() {
+				err = k8sClient.Delete(context.TODO(), generatorDpl, &client.DeleteOptions{})
+				Expect(err).Should(Succeed(), "Failed to delete logger deployment")
 
-					err = utils.WaitForReadyDeployment(k8sClient, dpl, defaultRetry, defaultTimeout)
-					Expect(err).Should(Succeed(), fmt.Sprintf("Failed to wait for ready querier deployment: %s", dpl.GetName()))
+				for _, dpl := range querierDpls {
+					err := k8sClient.Delete(context.TODO(), dpl, &client.DeleteOptions{})
+					Expect(err).Should(Succeed(), "Failed to delete querier deployment")
 				}
-
-				// Sleeping for the first interval so that the data is accurate for the new workload.
-				time.Sleep(scenarioCfgs.Samples.Interval)
```

Comment:
Why is that not needed anymore?

Reply:
This was a buffer to reduce the interference from a previous test in the current one. Since the tests are now run with the full setup and tear down cycle, it is no longer needed.